src/main/java/walle/WallE.java

```diff
@@ -0,0 +1,220 @@
+        printExitMessage();
+        reader.close();
+    }
+
+    private static void processInput(String[] params, ArrayList<Task> tasks, String userInput) {
+        try {
+            switch (params[0]) {
+            case "list":
+                printTaskList(listSize);
+                break;
+
+            case "mark":
+                markTask(params);
+                break;
+
+            case "unmark":
+                unmarkTask(params);
+                break;
+
+            case "todo":
+                addTodo(params, userInput);
+                break;
+
+            case "deadline":
+                addDeadline(params);
+                break;
+
+            case "event":
+                addEvent(params);
+                break;
+
+            case "delete":
+                int taskIndex = Integer.parseInt(params[1]);
+                deleteTask(taskIndex - 1); // 1 indexed
+                break;
+
+            default:
+                throw new WallEException("Invalid command");
+            }
+        } catch(IndexOutOfBoundsException e) {
+            printWithLineBreak("Invalid command, please ensure correct arguments.");
+        } catch (WallEException e) {
+            printWithLineBreak(e.getMessage());
+        }
+    }
+
+    private static void addTask(String userInput) {
+        tasks.add(new Task(userInput));
+        listSize++;
+        printWithLineBreak("added: " + userInput);
+    }
+
+    private static void deleteTask(int taskIndex) {
+        Task deletedTask = tasks.remove(taskIndex);
+        listSize--;
+        printWithLineBreak("deleted task: " + deletedTask.toString());
+    }
+
+    private static void addEvent(String[] params) {
+        int fromIndex = 0;
+        int toIndex = 0;
+
+        StringBuilder eventName = new StringBuilder();
+        StringBuilder startDate = new StringBuilder();
+        StringBuilder endDate = new StringBuilder();
+        int stringType = 0;
+
+        for (int i = 1; i < params.length; i++) {
+            if (params[i].equals("/from")) {
+                stringType = 1;
+            } else if (params[i].equals("/to")) {
+                stringType = 2;
+            } else {
+                if (stringType == 0) {
+                    if (eventName.isEmpty()) {
+                        eventName.append(params[i]);
+                    } else {
+                        eventName.append(" ").append(params[i]);
+                    }
+                } else if (stringType == 1) {
+                    if (startDate.isEmpty()) {
+                        startDate.append(params[i]);
+                    } else {
+                        startDate.append(" ").append(params[i]);
+                    }
+                } else {
+                    if (endDate.isEmpty()) {
+                        endDate.append(params[i]);
+                    } else {
+                        endDate.append(" ").append(params[i]);
+                    }
```

Comment:
This is very deep nesting and can be considered arrowhead code. Try to refactor the internal layers into separate methods.